### DISCUSSION TITLE
Lohnt sich ein Update

### DISCUSSION THREAD

**Discussion by musbach** [tbnobody/OpenDTU#2067] [General]
Hallo,
bei mir läuft derzeit openDTU zuverlässig. Lohnt sich ein Update (Firmware-Version 24.4.24) und wenn ja, welche Firmware-Dateien muss ich nehmen?

![grafik](https://github.com/tbnobody/OpenDTU/assets/20500637/baf5b329-4fa0-43f5-b7d0-6d1a3c79845f)

![grafik](https://github.com/tbnobody/OpenDTU/assets/20500637/e9486109-2cdd-49b4-bbca-673b80a187ee)

**Comment by Tiese** [comment]
Hallo @musbach,

mein Chef hat immer gesagt: "**_Wenn die Anlage läuft - finger weg!_**"

Schöne Grüße
Tiese

**Reply by ghost** [reply]
Kennt jeder IT-ler, die halten sich außer bei Sicherheitsupdates auch immer daran und  ist schon sooooo alt: 
"Never change a running system."

**Comment by Solarteur** [comment]
Hier

https://github.com/tbnobody/OpenDTU/releases

ist beschrieben, welche Änderungen es bei den einzelnen Releases gab - als Orientierung, ob sich ein Update für den einzelnen "lohnt", welche Fehler behoben und welche zusätzlichen Funktionen realisiert wurden.

Der Screenshot von @musbach zeigt, dass er die Datei **opendtu-generic_esp32.bin** für ein Update braucht.

**Reply by ghost** [reply]
Der Screenshot von @musbach zeigt auch, dass er die Datei opendtu-generic.bin auf dem ESP hat.
Vor dem Umstieg auf opendtu-generic_esp32.bin mal noch sicherheitshalber die Pin-Konfiguration und vielleicht auch die Einstellungen für ein nachfolgendes Wiederherstellen sichern.

**Comment by musbach** [comment]
Vielen Dank für die Antworten. Ich habe nun die beiden Dateien gesichrert. Was ist der Unterschied, wenn ich von opendtu-generic.bin auf opendtu-generic_esp32.bin umsteige? Das lese ich nicht aus den Release-Notes.

**Comment by ghost** [comment]
Steht auch nicht dort da.

Bei den vor-[kompilierten Dateien](https://www.opendtu.solar/firmware/flash_esp/) steht bei den opendtu-generic: veraltet.
Ich denke der größte Unterschied ist, du kannst Pin-Konfigurationen nur so benutzen, wie diese in der openDTU-Soft im ESP drin stehen, und es geht nur das NRF24 Modul.
Solltest du mal etwas zusätzlich oder anders anschließen wollen, könnte des scheitern. Hier könnte ein Entwickler mehr dazu sagen.
Am Olimex ESP32-POE dürfte die opendtu-generic auch nicht funktionieren. Und auch nicht an allen ESPs wo das CMT2300A Modul dran ist.
Die opendtu-generic ist  wohl auch nur für alle, die sich an die Standardverdrahtung halten, und /oder eine Leiterplatte in der Community-Variante benutzen.

Mit der opendtu-generic_esp32.bin bist du da freier und kannst verdrahten wie du willst, sofern die PIN-Eigenschaften der verschiedenen Pins beachtet werden (Nur Ein- oder Ausgang, Beides....)
Und, die ist so ziemlich für alle ESP32. Die ...V3 ist speziell für den ESP32, der einen USB-Controller OnBoard hat.
Die ...factory ist, für neue ESPs, bzw. wird benötigt, wen die momentane Version zu alt ist (23.xx.xx und davor). 

Wie du genau umsteigst, steht [hier](https://www.opendtu.solar/firmware/howto/migrate_generic/).

**Comment by musbach** [comment]
I brauche openDTU im Wesentlichen für evcc, wo die Leitung beider Wechselrichter (Kostal und Hoymiles) zusammengeführt wird. evcc steuert dann meine Wallbox, wenn genügend Überschuss vorhanden ist. Diese Kommunikation ist für mich wichtig und muss zukunftssicher sein.

**Comment by ghost** [comment]
Das musst du für dich selbst beurteilen. Schaue in die unten genannten Links mal rein.
In den [release notes](https://github.com/tbnobody/OpenDTU/releases) steht drin, was neu oder anders ist.
Die [Dokumentation](https://www.opendtu.solar/) ist auch sehr empfehlenswert. 
Was soll an einem Update nicht zukunftssicher sein?

**Comment by musbach** [comment]
Vielen Dank! Das Update hat problemlos geklappt!
